### PR TITLE
fix(BaseInputField): fix proper passing of className prop

### LINKED ISSUE #947: Feil: BaseInputField - hvis className prop blir tatt i bruk overskriver den jøkul-styling
**Feilbeskrivelse**
BaseInputField har støtte for prop-en className, men hvis den blir tatt i bruk overskriver man enkelt og greit "jkl-text-input__input".

**Forventet oppførsel**
Dette er sikkert et ønsket pattern på visse komponenter, men akkurat på denne som er så enkel så ønsker jeg at den setter `jkl-text-input__input ${className}`.

Slik at jeg har kan definere `input--first` `input--second` etc. for å få korrekt margins
image

Mulig dette egentlig er en feature-request og ikke en feil, men føler at eksponering av className er feil når bruk av den kun fjerner styling.

### PR DESCRIPTION
**Pull request by vegardsoerlie**
affects: @fremtind/jkl-text-input-react

closes #947 

## 📥 Proposed changes

PR-en åpner for at man kan legge til custom className ved siden av `jkl-text-input__input`.

## ☑️ Submission checklist

-   [x] I have read the [CONTRIBUTING](https://github.com/fremtind/jokul/blob/master/CONTRIBUTING.md) doc
-   [x] `yarn build` works locally with my changes
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: packages/text-input-react/src/BaseInputField.tsx [hunk 1]
```diff
@@ -40,11 +40,11 @@ function getWidthAsStyle(width?: string, maxLength?: number): CSSProperties | un
 }
 
 export const BaseInputField = forwardRef<HTMLInputElement, Props>(
-    ({ id, describedBy, invalid, maxLength, width, type = "text", ...rest }, ref) => (
+    ({ id, describedBy, invalid, maxLength, width, type = "text", className, ...rest }, ref) => (
         <input
             ref={ref}
             id={id}
-            className="jkl-text-input__input"
+            className={`jkl-text-input__input ${className}`}
```
Comment (piofinn) [inline packages/text-input-react/src/BaseInputField.tsx:47]: Bør nok være en ternary for `className` her, ellers vil du ende opp med "jkl-text-input__input undefined" når ikke `className` er gitt.

Comment (vegardsoerlie) [reply packages/text-input-react/src/BaseInputField.tsx:47]: Godt poeng, tenkte det var så liten task så glemte å tenke skikkelig 🤦‍♂️

File: packages/text-input-react/src/BaseInputField.tsx [hunk 2]
```diff
@@ -40,11 +40,11 @@ function getWidthAsStyle(width?: string, maxLength?: number): CSSProperties | un
 }
 
 export const BaseInputField = forwardRef<HTMLInputElement, Props>(
-    ({ id, describedBy, invalid, maxLength, width, type = "text", ...rest }, ref) => (
+    ({ id, describedBy, invalid, maxLength, width, type = "text", className, ...rest }, ref) => (
```
Comment (lfbergee) [inline packages/text-input-react/src/BaseInputField.tsx:43]:
```suggestion
    ({ id, describedBy, invalid, maxLength, width, type = "text", className = "", ...rest }, ref) => (
```

File: packages/text-input-react/src/BaseInputField.tsx [hunk 3]
```diff
@@ -40,11 +40,11 @@ function getWidthAsStyle(width?: string, maxLength?: number): CSSProperties | un
...
         <input
             ref={ref}
             id={id}
-            className="jkl-text-input__input"
+            className={`jkl-text-input__input ${className ? className : ""}`}
```
Comment (lfbergee) [inline packages/text-input-react/src/BaseInputField.tsx:47]:
```suggestion
            className={`jkl-text-input__input ${className}`}
```

File: packages/text-input-react/src/BaseInputField.tsx (hunk 2 above)
Comment (piofinn) [reply packages/text-input-react/src/BaseInputField.tsx:43]: Smart!

Comment (piofinn) [reply packages/text-input-react/src/BaseInputField.tsx]: Smart!

**Pull request merged**